src/reducers/notebook-reducer.js

```diff
@@ -43,6 +43,18 @@ function clearHistory(loadedState) {
   /* eslint-enable */
 }
 
+function updateAppMessages(state, messageObj) {
+  const message = Object.assign({}, messageObj)
+  delete message.type
+  if (message.when === undefined) {
+    message.when = (new Date()).toString()
+  }
+  if (message.details === undefined) message.details = message.message
+  const nextState = Object.assign({}, state)
+  nextState.appMessages = nextState.appMessages.slice()
+  return addAppMessageToState(nextState, message)
+}
```

Comment:
these checks for message.when and message.details should be part of the message constructor rather than the reducer.